datahub-web-react/src/utils/formatter/index.ts

```diff
@@ -1,20 +1,29 @@
-const intlFormat = (num) => {
-    return new Intl.NumberFormat().format(Math.round(num * 10) / 10);
-};
+const NumMap = {
+    billion: {
+        value: 1000000000,
+        symbol: 'B',
+    },
+    million: {
+        value: 1000000,
+        symbol: 'M',
+    },
+    thousand: {
+        value: 1000,
+        symbol: 'K',
+    },
+} as const;
 
-export const countFormatter: (num: number) => string = (num: number) => {
-    if (num >= 1000000000) {
-        return `${intlFormat(num / 1000000000)}B`;
-    }
-    if (num >= 1000000) {
-        return `${intlFormat(num / 1000000)}M`;
-    }
+const isBillions = (num: number) => num >= NumMap.billion.value;
+const isMillions = (num: number) => num >= NumMap.million.value;
+const isThousands = (num: number) => num >= NumMap.thousand.value;
 
-    if (num >= 1000) return `${intlFormat(num / 1000)}K`;
+const intlFormat = (num: number) => new Intl.NumberFormat().format(Math.round(num * 10) / 10);
 
-    return intlFormat(num);
-};
+export const needsFormatting = (num: number) => isThousands(num);
```

Comment:
Started down a path of trying to check equality of 3500 vs 3.5K and with internationalization in play (Intl.NumberFormat) I don't feel like parsing here is going to be super reliable on machines with other languages. Future improvement to make for sure.